update docs w.r.t. cos7, alma8 & alma9 (#2377)
conda-forge/conda-forge.github.io: docs/maintainer/knowledge_base.md

Some dependencies are so close to the system that they are not packaged with conda-forge.
These dependencies have to be satisfied with _Core Dependency Tree_ (CDT) packages.

A CDT package consists of repackaged CentOS/AlmaLinux binaries from the appropriate version,
either 7, 8 or 9 depending on user choice and platform. We manage the build of CDT
packages using a centralized repo, [conda-forge/cdt-builds](https://github.com/conda-forge/cdt-builds),
as opposed to generating feedstocks for them. (Note that historically we did use feedstocks but this

> practice has been deprecated.) To add a new CDT, make a PR on the

Reviewer: ```suggestion
practice has been deprecated). To add a new CDT, make a PR on the
```

> <a id="using-centos-7"></a>

Author: Not sure if we want to touch the anchors here...

Author: CC @jaimergp

Reviewer: I'd leave it there, since it's still CentOS 7 related-ish, in case someone was linking there somewhere.

```yaml
c_stdlib_version:          # [linux]
  - "2.28"                 # [linux]
```

This covers the headers/library present at build-time, and will also create
a corresponding run-export on the `__glibc` virtual package.

> By default, conda-forge will default to old sysroots on new images, meaning that

Reviewer 2: ```suggestion
By default, conda-forge build infrastructure uses old sysroots on new images, meaning that
```

This is my only suggestion. Otherwise, LGTM.
